test/run.d

```diff
@@ -133,6 +130,45 @@ void ensureToolsExists()
         resultsDir.buildPath(dir).mkdirRecurse;
 }
 
+/// A single target to execute.
+immutable struct Target
+{
+    /**
+    The filename of the target.
+
+    Might be `null` if the target is not for a single file.
+    */
+    string filename;
+
+    /// The arguments how to execute the target.
+    string[] args;
+
+    /// Returns: the normalized test name
+    static string normalizedTestName(string filename)
+    {
+        return filename
+            .absolutePath
+            .dirName
+            .baseName
+            .buildPath(filename.baseName);
+    }
+
+    string normalizedTestName()
+    {
+        return Target.normalizedTestName(filename);
+    }
+
+    /// Returns: `true` if the test exists
+    bool exists()
+    {
+        // This is assumed to be the `unit_tests` target which always exists
+        if (filename.empty)
+            return true;
+
+        return testPath(normalizedTestName).exists;
```

Comment:
`testPath` already normalizes.

Reply:
It's not the same thing. `normalizedTestName` does what this line did: https://github.com/dlang/dmd/pull/8528/files#diff-060ed063c39ccc738bd0e384c097973cL67.

Look at the bottom of this failing CircleCI test to see why it's necessary: https://circleci.com/gh/dlang/dmd/25742.